src/diff.ts

```diff
@@ -0,0 +1,82 @@
+import parseDiff, { AddChange } from 'parse-diff'
+import { LcovFile } from 'lcov-parse'
+import { ContextInfo, Octokit } from './types'
+import { percentString } from './util'
+
+interface Props {
+  lcovFiles: LcovFile[]
+  octokit: Octokit
+  context: ContextInfo
+}
+
+export const analyzeDiffCoverage = async ({
+  lcovFiles,
+  octokit,
+  context
+}: Props) => {
+  const result = await octokit.rest.repos.compareCommitsWithBasehead({
+    owner: context.owner,
+    repo: context.repo,
+    basehead: `${context.compareSha}...${context.sha}`,
+    mediaType: {
+      format: 'diff'
+    }
+  })
+
+  let diff = parseDiff(result.data.toString())
+
+  diff = diff.filter(file =>
+    lcovFiles.find(lcovFile => lcovFile.file === file.to)
+  )
+
+  const fileChanges: Map<string, AddChange[]> = new Map()
+
+  for (const file of diff) {
+    for (const chunk of file.chunks) {
+      for (const change of chunk.changes) {
+        if (change.type === 'add') {
+          const changes = fileChanges.get(file.to!) ?? []
+          changes.push(change)
+          fileChanges.set(file.to!, changes)
+        }
+      }
+    }
+  }
+
+  let newLinesCovered = 0
+  let totalLines = 0
+
+  for (const lcovFile of lcovFiles) {
+    const changes = fileChanges.get(lcovFile.file)
+    if (!changes) {
+      continue
+    }
+
+    for (const detail of lcovFile.lines.details) {
+      const inChanges = changes.find(change => change.ln === detail.line)
+
+      if (inChanges) {
+        if (detail.hit > 0) {
+          newLinesCovered++
+        }
+        totalLines++
+      }
+    }
+  }
+
+  const success = totalLines > 0
```

Comment:
We should probably have a function the returns the patch analysis without a commit status. 
This would  make the whole thing more maintainable.
Have that function return  a {linesFound, linesHit} object.